metrics/sources/kubelet/kubelet.go

```diff
@@ -132,25 +157,12 @@ func (this *kubeletMetricsSource) decodeMetrics(c *cadvisor.ContainerInfo) (stri
 			}
 		}
 
+		// No Kubernetes metadata so treat this as a system container.
 		if cName == "" || ns == "" || podName == "" {
```

Comment:
So I think the extraction of container, pod & namespace looks right to me & I think we should trust that this will work. I can't see any problems with it.